tests/poet/test_poet_ui.py

```diff
@@ -0,0 +1,69 @@
+import requests
+
+from urllib.error import HTTPError
+from selenium.common.exceptions import NoSuchElementException
+
+import pytest
+
+from pages.poet.home import Home
+
+import os
+
+"""
+Verifies POET ui
+Latest update on May 21st, 2021
+"""
+
+
+def test_poet_ui(selenium, headers_data):
+    github_user = os.environ.get("GITHUB_USER")
+    github_password = os.environ.get("GITHUB_PASSWORD")
+
+    git_repo_url = os.environ.get("GIT_REPO_URL")
+
+    try:
+        requests.get(git_repo_url, headers=headers_data)
+
+    except HTTPError as h_e:
+        # Return code 404, 501, ... for incorrect repo url
+        pytest.fail(f"HTTP Error {h_e.code}: incorrect repo url {git_repo_url}")
+
+    else:
+        # launches gitpod for openstax/poet test repo and logs in
+        home = Home(selenium, git_repo_url).open()
+
+        home.click_github_login_button()
+
+        selenium.switch_to.window(selenium.window_handles[1])
+
+        home.fill_user_field(github_user)
+        home.fill_passwd_field(github_password)
```

Comment:
Is the idea / intent that these tests would get run with a dedicated test GH user with limited perms or an individual's real credentials (the latter would only work here if the user doesn't have MFA enabled on their account I think)?

Reply:
What I was thinking is to have an auto smoke test for poet everytime we release a new extension to registry. Basically, new extension is in registry, this set of tests would automatically kick off and test the poet and make sure that it runs. Something similar, what we had in cnx-automation for cnx webview. Just an idea and also for me to practice writing auto-tests :) But it would be good to have it.

Reply:
Imagine it as a ui smoke test of poet extension

Reply:
👍